File: backend/Domain/Entities/Property.cs
namespace Domain.Entities
{
    public class Property
    {
        public string Id { get; set; } = string.Empty;
        public string IdProperty { get; set; } = string.Empty;
        public string Name { get; set; } = string.Empty;
        public string Address { get; set; } = string.Empty;
        public decimal Price { get; set; }
        public string CodeInternal { get; set; } = string.Empty;
        public int Year { get; set; }
        public string IdOwner { get; set; } = string.Empty;

        // Relaciones
        public Owner? Owner { get; set; }
        public ICollection<PropertyImage>? Images { get; set; }
        public ICollection<PropertyTrace>? Traces { get; set; }
    }
}

File: frontend/src/core/entities/Property.ts
// src/core/entities/Property.ts
export interface PropertyImage {
  idPropertyImage: string;
  idProperty: string;
  file: string;
  enabled: boolean;
}

export interface PropertyTrace {
  idPropertyTrace: string;
  dateSale: string;
  name: string;
  value: number;
  tax: number;
  idProperty: string;
}

export interface Owner {
  idOwner: string;
  name: string;
  address: string;
  photo: string;
  birthday: string;
}

export interface Property {
  idProperty: string;
  name: string;
  address: string;
  price: number;
  codeInternal: string;
  year: number;
  description?: string;
  owner?: Owner;
  images?: PropertyImage[];
  traces?: PropertyTrace[];
}
